src/gp_activetable.c

```diff
@@ -198,15 +202,78 @@ object_access_hook_QuotaStmt(ObjectAccessType access, Oid classId, Oid objectId,
 {
 	if (prev_object_access_hook) (*prev_object_access_hook)(access, classId, objectId, subId, arg);
 
-	/* if is 'drop extension diskquota' */
-	if (classId == ExtensionRelationId && access == OAT_DROP)
+	if (access == OAT_DROP)
 	{
-		if (get_extension_oid("diskquota", true) == objectId)
+		switch (classId)
 		{
-			invalidate_database_rejectmap(MyDatabaseId);
-			diskquota_stop_worker();
+			case ExtensionRelationId:
+				/* if is 'drop extension diskquota' */
+				if (get_extension_oid("diskquota", true) == objectId)
+				{
+					invalidate_database_rejectmap(MyDatabaseId);
+					diskquota_stop_worker();
+				}
+				return;
+
+			case NamespaceRelationId:
+			case AuthIdRelationId:
+			case TableSpaceRelationId: {
```

Comment:
Is this logic valid on segments?

Reply:
No, it is not. So I add `if (Gp_role != GP_ROLE_DISPATCH) return;`

Reply:
> if (Gp_role != GP_ROLE_DISPATCH) return;

But up to this line, all code is executed on segments too! That's why I suggest raising this condition as high as possible.

Reply:
It is necessary to call `invalidate_database_rejectmap` on segments when the `drop extension diskquota` query is executed. So I cannot place the check of Gp_role above the switch